internal/resources/route/route.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2024 The KubeLB Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package route
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/go-logr/logr"
+
+	kubelbk8ciov1alpha1 "k8c.io/kubelb/api/kubelb.k8c.io/v1alpha1"
+	"k8c.io/kubelb/internal/kubelb"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	gwapiv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+type Subresources struct {
+	Services        []corev1.Service           `json:"services,omitempty"`
+	ReferenceGrants []gwapiv1a2.ReferenceGrant `json:"referenceGrants,omitempty"`
+}
+
+func CreateRouteForResource(ctx context.Context, _ logr.Logger, client ctrlclient.Client, resource unstructured.Unstructured, subresources Subresources, namespace string) error {
+	generateRoute := GenerateRoute(resource, subresources, namespace)
+	return CreateUpdateRoute(ctx, client, generateRoute)
+}
+
+func GenerateRoute(resource unstructured.Unstructured, resources Subresources, namespace string) kubelbk8ciov1alpha1.Route {
+	return kubelbk8ciov1alpha1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      string(resource.GetUID()),
+			Namespace: namespace,
+			Labels: map[string]string{
+				kubelb.LabelOriginNamespace:    resource.GetNamespace(),
+				kubelb.LabelOriginName:         resource.GetName(),
+				kubelb.LabelOriginResourceKind: resource.GetObjectKind().GroupVersionKind().GroupKind().String(),
+				kubelb.LabelManagedBy:          kubelb.LabelControllerName,
+			},
+		},
+		Spec: kubelbk8ciov1alpha1.RouteSpec{
```

Comment:
For reviewer:

`Endpoints` have been intentionally ignored at the moment since in a follow up PR I'd like to change the way node IPs are propagated from CCM to KubeLB manager; it should be a single CRD instead of the CCM updating all the LoadBalancer and Routes without taking into account what's managed and not managed by KubeLB.

For example, if someone creates a `LoadBalancer` in the tenant cluster by hand, create CR manually, it's endpoints will automatically be updated by the CCM which is incorrect.